src/jinja2/utils.py

```diff
@@ -653,6 +697,60 @@ def htmlsafe_json_dumps(
     )
 
 
+@lru_cache(100)
+def get_wrapped_escape_class(custom_escape: Callable[[Any], str]) -> Type[Markup]:
+    """
+    Use a simple escape function to generate a wrapped Markup class
+
+    This class uses the given ``custom_escape`` function to escape
+    the value andat the same time makes sure that no already escaped
+    string is escaped again.
+
+    The returned class is a subclass of :class:`markupsafe.Markup`,
+    so it represents a complete ``str`` subclass that is marked as
+    safe
+
+    :param custom_escape: The function that escapes the objects to a str
+
+    :returns an Markup class using this escape function
+
+    .. versionadded:: 3.0
+    """
+
+    class MarkupWrapper(Markup):
+        """
+        Make sure that the custom escape function is used
+        """
+
+        @classmethod
+        def get_unwrapped_escape(cls):
+            # Needed for test
+            return custom_escape
+
+        @classmethod
+        def escape(cls, s):
+            """
+            Make sure the custom escape function does not escape
+            already escaped strings
+            Also make sure the escaped string is marked as escaped
+            with the correct class
+
+            If the object has an ``__html__`` method, it is called
+            and the return value is assumed to already be safe for HTML
+            / resp what ever is escaped currently.
+            The HTML Attribue shall be understood simply as
+                "it is a safe string"
+            """
+            if hasattr(s, "__html__"):
+                return cls(s.__html__())
+            return cls(custom_escape(s))  # noqa: B902
+
+    return MarkupWrapper
+
+
+get_wrapped_escape_class
```

Comment:
👀

Reply:
Aehm, how did that ended up there...
Strange that no linter found it, as it is completely without any function.